### PR TITLE
Ensure that update operations are retried

### PR DESCRIPTION
**Pull request by skitt**
All Kubernetes update operations should be retried on conflicts,
i.e. the object being changed has changed on the server since it was
retrieved.

Fixes: #255
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
TODO:

[ ] Add a full set of tests
[ ] Handle the Submariner CR and SCC

File: pkg/utils/createorupdate.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+			clusterRole.ResourceVersion = existingClusterRole.ResourceVersion
+			// Potentially retried
+			_, err = clientSet.RbacV1().ClusterRoles().Update(clusterRole)
+			if err != nil {
+				return fmt.Errorf("failed to update pre-existing cluster role %s : %v", clusterRole.Name, err)
+			}
+			return nil
+		})
+		return false, retryErr
+	}
```
Comment (tpantelis) [inline pkg/utils/createorupdate.go]: I think we should use ` controllerutil.CreateOrUpdate` as is done [here](https://github.com/submariner-io/submariner-operator/blob/master/pkg/controller/submariner/submariner_controller.go#L178) but perhaps that should be in another PR.

Comment (skitt) [reply pkg/utils/createorupdate.go]: I tried to do that, but `controllerutil` works in the context of a controller (with a plan client and context), it doesn’t work with a clientSet. I imagine it’s possible to get a client and context from a clientSet, but I haven’t figured out how (yet).

Comment (tpantelis) [reply pkg/utils/createorupdate.go]: right - forgot about that. We can look into that later. It's a little harder to do it generically with client sets.
